cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectApi.java

```diff
@@ -212,4 +215,37 @@ public KafkaConnectApiImpl(Vertx vertx) {
                 .end();
         return result;
     }
+
+    @Override
+    public Future<List<ConnectorPlugin>> listConnectorPlugins(String host, int port) {
+        Future<List<ConnectorPlugin>> result = Future.future();
+        HttpClientOptions options = new HttpClientOptions().setLogActivity(true);
+        String path = "/connector-plugins";
+        vertx.createHttpClient(options)
+                .get(port, host, path, response -> {
+                    response.exceptionHandler(error -> {
+                        result.fail(error);
+                    });
+                    if (response.statusCode() == 200) {
+                        response.bodyHandler(buffer -> {
+                            ObjectMapper mapper = new ObjectMapper();
+
+                            try {
+                                result.complete(Arrays.asList(mapper.readValue(buffer.getBytes(), ConnectorPlugin[].class)));
+                            } catch (IOException e)  {
+                                log.warn("Failed to parse list of connector plugins");
+                                result.fail(e);
```

Comment:
I wonder if you should use `ConnectRestException` here. Until now `ConnectRestException` has been for lower level errors in handling the HTTP. Here the HTTP worked but returned something that wasn't valid JSON, or was not the expected JSON.